packages/rum-core/src/domain/rumEventsCollection/view/viewMetrics/trackFirstHidden.ts

```diff
@@ -2,40 +2,42 @@ import type { RelativeTime } from '@datadog/browser-core'
 import { addEventListeners, DOM_EVENT } from '@datadog/browser-core'
 import type { RumConfiguration } from '../../../configuration'
 
-let trackFirstHiddenSingleton: { timeStamp: RelativeTime } | undefined
-let stopListeners: (() => void) | undefined
+let trackFirstHiddenSingleton: { timeStamp: RelativeTime; stop: () => void } | undefined
 
 export function trackFirstHidden(configuration: RumConfiguration, eventTarget: Window = window) {
   if (!trackFirstHiddenSingleton) {
     if (document.visibilityState === 'hidden') {
       trackFirstHiddenSingleton = {
         timeStamp: 0 as RelativeTime,
+        stop: reset,
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏mixing a `stop()` method with singletons scares me a bit, because `stop()` could be called while the singleton might still be used in a totally different place.

In this particular case, we know that it's not the case because everything using the singleton call stop() at the same time, but still, I don't think it is a good design.